### PR TITLE
gh-103300: Fix `Popen.wait()` deadlock in patchcheck.py

### PR DESCRIPTION
**Pull request by arhadthedev**
Use `Popen.communicate()` as recommended by https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait.

* Issue: gh-103300

### CODE REVIEW

**Comment by arhadthedev** [comment]
`test_tools` doesn't cover patchcheck so no test attached.

File: Misc/NEWS.d/next/Tools-Demos/2023-04-06-06-19-39.gh-issue-103300.nnE7Be.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+:source:`Tools/patchcheck/patchcheck.py` no longer deadlocks trying to
```
Comment (sobolevn) [inline Misc/NEWS.d/next/Tools-Demos/2023-04-06-06-19-39.gh-issue-103300.nnE7Be.rst]: Tools do not need NEWS, AFAIK.

**Review by sobolevn — COMMENTED** [review]
LGTM 👍

**Comment by sunmy2019** [comment]
LGTM. I have encountered such type of bugs before.

**Pull request merged**